plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -372,7 +372,9 @@ function perflab_install_and_activate_plugin( string $plugin_slug, array &$proce
 		// Replace new Plugin_Installer_Skin with new Quiet_Upgrader_Skin when output needs to be suppressed.
 		$skin     = new WP_Ajax_Upgrader_Skin( array( 'api' => $plugin_data ) );
 		$upgrader = new Plugin_Upgrader( $skin );
-		$result   = $upgrader->install( $plugin_data['download_link'] );
+		// Remove the version number from the link to download the latest plugin version.
+		$download_link = (string) preg_replace( '#(\/plugin\/[^\/]+)\.\d+\.\d+\.\d+(?:-\w+)?\.zip#', '$1.zip', $plugin_data['download_link'] );
```

Comment:
I'm not a fan of regexing this from the `download_link`. Is there no externally controlled place (API responses or elsewhere) that exposes the format we want here?

Reply:
Seeing comments like https://github.com/WordPress/performance/pull/1693#discussion_r1856636899 confirms my above concern: Using a regex for this is error-prone. Is there no way to get the actual URL from somewhere?

Reply:
You’re right; relying on regex for this isn’t ideal. The `plugins_api()` function, when called with the `plugin_information` action, provides a `versions` field in the response that seems to include the URLs in the required format. Here's an example of the response:

```json
{
    ...
    "versions": {
        "0.1.0": "https://downloads.wordpress.org/plugin/optimization-detective.0.1.0.zip",
        "0.1.1": "https://downloads.wordpress.org/plugin/optimization-detective.0.1.1.zip",
        ...
        "trunk": "https://downloads.wordpress.org/plugin/optimization-detective.zip"
    },
    ...
}
```

We could use the `trunk` key to get the required URL in most cases. However, there are scenarios where the `trunk` key doesn’t exist in the `versions` field. In those cases, the `download_link` itself is typically already in the desired format without a version or tag.

That said, implementing this approach would require making multiple API calls—one for each plugin—which again might not be ideal.

Reply:
Don't we only need the plugin download URL if we actually want to download the plugin? I think making an extra request for that purpose makes sense. We shouldn't make all these requests for all the plugins when we cache the information. But for the moment when someone clicks "Activate" and we have to install that plugin, I think it's reasonable to make the request. Even if the plugin also requires dependencies, making individual requests for these should not be an issue since realistically that shouldn't be more than 2-3 plugins.

This solution would be more reliable than us assuming things about the download URL and using a regex.

Reply:
That makes sense to me.